src/compiler/checker.ts

```diff
@@ -9384,8 +9455,13 @@ namespace ts {
             // Decorators will not have `excludeArgument`, as their arguments cannot be contextually typed.
             // Tagged template expressions will always have `undefined` for `excludeArgument[0]`.
             if (excludeArgument) {
+                if (signature.thisType && calleeNode) {
+                    if (excludeCallee === false) {
+                        inferTypes(context, checkExpressionWithContextualType(calleeNode, signature.thisType, inferenceMapper), signature.thisType);

```

Comment:
Just use `checkExpression(calleeNode)` here, don't supply a contextual type. Actually, we may want to consider caching that type so we don't have to do a full type check on the node again (we already did that in order to get here in the first place).